pkg/util/create_or_update.go

```diff
@@ -88,3 +101,32 @@ func CreateOrUpdate(client dynamic.ResourceInterface, obj *unstructured.Unstruct
 
 	return result, nil
 }
+
+// CreateAnew creates a resource, first deleting an existing instance if one exists.
+func CreateAnew(client dynamic.ResourceInterface, obj runtime.Object) error {
+	toCreate := &unstructured.Unstructured{}
+	err := scheme.Scheme.Convert(obj, toCreate, nil)
+	if err != nil {
+		return err
+	}
+
+	return retry.OnError(backOff, apierrors.IsAlreadyExists, func() error {
```

Comment:
Is there any point in retrying if `Create()` returns an error other than “already exists”?

Reply:
OK, for anyone else following along, @tpantelis pointed out to me that the function is only retried if the error is accepted by the function given as the second argument to `OnError`, _i.e._ `IsAlreadyExists`.

Reply:
I refactored it to use wait.ExponentialBackoff as OnError doesn't really add value.